src/components/core/geoChart/geoChartDataLayers.ts

```diff
@@ -46,17 +46,21 @@ function createPushpinSizeOptions(geoData: IGeoData): mapboxgl.Expression | numb
     // The mouseenter event uses queryRenderedFeatures to determine when the mouse is touching a feature
     // And queryRenderedFeatures is not supporting nested object in expression
     // https://github.com/mapbox/mapbox-gl-js/issues/7194
-    const sizeOptions: mapboxgl.Expression = ["step", ["get", "pushpinRadius"], DEFAULT_PUSHPIN_SIZE_VALUE];
+
+    const stepSizeOptions: mapboxgl.Expression = [
+        "step",
+        ["get", "pushpinRadius"],
+        DEFAULT_PUSHPIN_SIZE_VALUE,
+    ];
     for (let index = 0; index < sizesCount; index++) {
         let stepValue = sizeMin + sizeStep * index;
         stepValue = parseFloat(stepValue.toFixed(2));
         if (stepValue > sizeMax) {
             stepValue = sizeMax;
         }
-        sizeOptions.push(stepValue, DEFAULT_PUSHPIN_SIZE_SCALE[index]);
+        stepSizeOptions.push(stepValue, DEFAULT_PUSHPIN_SIZE_SCALE[index]);
     }
-
-    return sizeOptions;
+    return ["case", ["==", ["get", "pushpinRadius"], null], DEFAULT_PUSHPIN_SIZE_VALUE, stepSizeOptions];
```

Comment:
should we care about `undefined`, `NaN`?

Reply:
Currently, data in feature of Mapbox is `NaN`. And I tested, filter Mapbox just works with comparing  `null`  value.